visualization/dashboard/src/MLIDashboard/Controls/FilterEditor.tsx

```diff
@@ -81,131 +163,176 @@ export class FilterEditor extends React.PureComponent<IFilterEditorProps, IFilte
         {
             key: FilterMethods.lessThan,
             text: localization.Filters.lessThanComparison
+        },
+        {
+            key: FilterMethods.inTheRangeOf,
+            text: localization.Filters.inTheRangeOf
         }
     ];
     private _isInitialized = false;
 
     constructor(props: IFilterEditorProps) {
         super(props);
-        this.state = _.cloneDeep(this.props.initialFilter);
+        this.state = {openedFilter: this.props.initialFilter};
         this._leftSelection = new Selection({
             selectionMode: SelectionMode.single,
             onSelectionChanged: this._setSelection
           });
         this._leftSelection.setItems(this.leftItems);
-        this._leftSelection.setKeySelected(this.extractSelectionKey(this.props.initialFilter.column), true, false);
         this._isInitialized = true;
     }
 
-    public render(): React.ReactNode {
-        const selectedMeta = this.props.jointDataset.metaDict[this.state.column];
-        const numericDelta = selectedMeta.treatAsCategorical || selectedMeta.featureRange.rangeType === RangeTypes.integer ?
-            1 : (selectedMeta.featureRange.max - selectedMeta.featureRange.min)/10;
-        const isDataColumn = this.state.column.indexOf(JointDataset.DataLabelRoot) !== -1;
-        let categoricalOptions: IComboBoxOption[] = [];
-        if (selectedMeta.treatAsCategorical) {
-            // Numerical values treated as categorical are stored with the values in the column,
-            // true categorical values store indexes to the string values
-            categoricalOptions = selectedMeta.isCategorical ? 
-                selectedMeta.sortedCategoricalValues.map((label, index) => {return {key: index, text: label}}) :
-                selectedMeta.sortedCategoricalValues.map((label) => {return {key: label, text: label.toString()}})
+    componentDidUpdate(props:IFilterEditorProps, prevVal: IFilterEditorState) {
```

Comment:
ComponentDidUpdate is not always a good idea, it seems like this can all be done without it, if we have the cohortEditor own a state property called editingFilter which can be undefined. If its undefined, the placeholder is shown, otherwise, the appropriate right panel controls. Again, I think this will be easier to handle if all the cohort editing is handled in a single class